File: pset6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    # Count characters, words and sentences
    chars = count_chars(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Calculate the score using the formula
    score = formula(chars, words, sentences)

    # Print the result
    print_result(score)


# Counting the characters of text
def count_chars(text):
    number = 0
    for char in text:
        if char.isalpha():
            number += 1
    return number


# Counting the words of text
def count_words(text):
    number = 1
    for char in text:
        if char == " ":
            number += 1
    return number


# Counting the sentences of text
def count_sentences(text):
    number = 0
    for char in text:
        if char == "." or char == "?" or char == "!":
            number += 1
    return number


# Calculating the formula to find the score
def formula(c, w, s):
    L = c / w * 100
    S = s / w * 100
    score = round((0.0588 * L) - (0.296 * S) - 15.8)
    return score


# Print the correct text based on the score
def print_result(score):
    if score < 1:
        print("Before Grade 1")
    elif score > 16:
        print("Grade 16+")
    else:
        print(f"Grade {score}")


main()    

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

int calculate_grade(string text);

int main(void)
{
    // Promt user for text
    string text = get_string("Text: ");

    // Calculate grade of text
    int grade = calculate_grade(text);

    // Print correct grade of text
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

// Calculates the readability of the prompted text
int calculate_grade(string text)
{
    int letters = 0;
    int sentences = 0;
    int words = 1;

    // Count letters, sentences and words
    for (int i = 0; i < strlen(text); i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            letters++;
        }
        else if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sentences++;
        }
        else if (text[i] == 32)
        {
            words++;
        }
    }

    // Final equation to calculate readability
    float L = letters / (float) words * 100;
    float S = sentences / (float) words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    return round(index);
}